<launch>
    <!-- 
        Note! For  model and config the launch files expects 
        the files to be in the folder:

        ..../spot_micro_perception/data/networks/
        While the labels are expected to be in 
        ..../spot_micro_perception/data/labels/
     -->

    <arg name="display_detections" default="false"/>  
    <arg name="track_mode" default="true"/>
    <arg name="configs_folder" default="$(find spot_micro_perception)/data"/>
    
    <arg name="yaml_file" default="default.yaml"/>
    <arg name="track_class" default="0"/>

    <!-- Launch camera publisher node -->
    <node pkg="spot_micro_perception" 
    type="cameraPublisher.py" 
    name="camera_publisher_node" 
    output="screen"/>
    <!-- 
        Launches the object detection manager node, this node  tries
        to recognize the architecture of the model and load it with the 
        appropriate tools. As of now only Darknet is supported but after
        refactoring to ROS Humble support will be improved.
    -->
    <node pkg="spot_micro_perception"
    required="true"
    type="spotMicroNetworkManager.py" 
    name="network_manager" 
    output="screen">
        <param name="display_detections" value="$(arg display_detections)"/>
        <param name="configs_folder" value="$(arg configs_folder)"/>
        <param name="config_file" value="$(arg configs_folder)/$(arg yaml_file)"/>
    </node>


    <!-- 
        Launches the object tracker node. When a detection is made
        the robot moves in such a way to always have the object
        at the center of it's vision minus some tolerance interval.
    -->
    <node pkg="spot_micro_perception" 
        type="spotMicroPerceptionMove.py" 
        name="perception_track" 
        output="screen">
        <param name="tracking_id" value="$(arg track_class)"/>
    </node>

</launch>
